### PR TITLE
Experimental: mungegithub/publisher: use filter-handler instead of cherry-picks

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: mungegithub/Dockerfile-publisher [hunk 1]
```diff
@@ -15,7 +15,7 @@
 FROM google/debian:jessie
 MAINTAINER Chao Xu <xuchao@google.com>
 RUN apt-get update \
- && apt-get install -y -qq git=1:2.1.4-2.1+deb8u2 \
+ && apt-get install -y -qq git=1:2.1.4-2.1+deb8u4 \
```
Comment (rmmh) [inline mungegithub/Dockerfile-publisher]: Can you just remove the version pinning? It's only going to continue breaking in the future.

File: mungegithub/Dockerfile-publisher [hunk 2]
```diff
@@ -15,7 +15,7 @@
...
  && apt-get install -y -qq mercurial=3.1.2-2+deb8u3 \
  && apt-get install -y -qq ca-certificates wget jq vim tmux \
  && wget https://storage.googleapis.com/golang/go1.8.1.linux-amd64.tar.gz \
```
Comment (rmmh) [inline mungegithub/Dockerfile-publisher]: could use go1.8.3

File: mungegithub/publisher/README.md [hunk 3]
```diff
@@ -26,17 +26,17 @@ The robot is also responsible to update the `Godeps/Godeps.json` and the `vendor
 
 Currently we don't have tests for the robot. It relies on manual tests:
 
-* Fork the repos you are going the publish. Run [fetch-all-latest-and-push.sh](util/fetch-all-latest-and-push.sh) to update the branches of your repos.
+* Fork the repos you are going the publish. Run [fetch-all-latest-and-push.sh](fetch-all-latest-and-push.sh) to update the branches of your repos.
 
-* Change `config.organization` to your github username in `mungegithub/publisher/deployment/kubernetes/configmap.yaml`
```
Comment (rmmh) [inline mungegithub/publisher/README.md]: This structure was for consistency with other deployments. Moving the configmap will break `make push_config`.

File: mungegithub/mungers/publish_scripts/publish_template.sh [hunk 4]
```diff
@@ -60,10 +60,12 @@ SCRIPT_DIR=$(dirname "${BASH_SOURCE}")
 source "${SCRIPT_DIR}"/util.sh
 
 git fetch origin
+git rebase --abort || true
+git reset --hard
+git clean -f -f -f
```
Comment (rmmh) [inline mungegithub/mungers/publish_scripts/publish_template.sh]: `git clean -ffdx`?

File: mungegithub/mungers/publisher.go [hunk 5]
```diff
@@ -354,7 +422,7 @@ func (p *PublisherMunger) publish() error {
 		for _, branchRule := range repoRules.srcToDst {
```
Comment (rmmh) [inline mungegithub/mungers/publisher.go]: If you add this, it's possible to test the script in dry run, skipping only the "git push" at the end:

		if p.githubConfig.DryRun {
			continue
		}

File: mungegithub/Dockerfile-publisher (hunk 1 above)
Comment (sttts) [inline mungegithub/Dockerfile-publisher]: This is intentional. We depend quite a bit on git's behaviour and prefer to have it break than have wrong results.

File: mungegithub/publisher/README.md (hunk 3 above)
Comment (sttts) [inline mungegithub/publisher/README.md]: Only fixing what the deploy.sh script expects: a config map

File: mungegithub/mungers/publish_scripts/initialize_repos.sh [hunk 6]
```diff
@@ -21,18 +21,36 @@ set -o errexit
...
+ORG="${ORG:-kubernetes}"
 
 mkdir -p "${GOPATH}"/src/k8s.io
 cd "${GOPATH}"/src/k8s.io
 
-# to restore the dependencies
-git clone "https://github.com/kubernetes/kubernetes"
-pushd kubernetes
-    ./hack/godep-restore.sh
+# Install godep
```
Comment (caesarxuchao) [inline mungegithub/mungers/publish_scripts/initialize_repos.sh]: Why the godep installation from the Dockerfile to initialize_repos.sh?

File: mungegithub/mungers/publish_scripts/publish_template.sh (hunk 4 above)
Comment (caesarxuchao) [reply mungegithub/mungers/publish_scripts/publish_template.sh]: What does the third `-f` do? The man page doesn't say.

rmmh's suggestion looks better.

File: mungegithub/mungers/publish_scripts/util.sh [hunk 7]
```diff
@@ -35,88 +35,91 @@ set -o pipefail
 # 6. for each commit, cherry-pick it (which will keep authorship) into "target-branch"
 # 7. update metadata files indicating which commits we've sync'ed to
 sync_repo() {
```
Comment (caesarxuchao) [inline mungegithub/mungers/publish_scripts/util.sh]: Could you update the comment regarding the workflow?

**Comment by sttts** [comment]
@caesarxuchao this PR was an experiment for an alternative implementation to cherry-picks. But I have my doubt that it makes it much easier. Please take a look at https://github.com/kubernetes/test-infra/pull/4134 instead.

**Pull request closed without merging**